tests/rados/test_objectstore_block.py

```diff
@@ -0,0 +1,223 @@
+            )
+            obj_dump = json.loads(obj_dump_)
+            obj_dump_stats = obj_dump["stat"]
+            obj_dump_extents = obj_dump["onode"]["extents"]
+        except (KeyError, ValueError) as e:
+            log.error(
+                f"Object dump json for {object_name} could not be fetched correctly"
+            )
+            log.error(f"{e.__doc__}")
+            log.exception(e)
+
+        log.debug(f"Object dump for {object_name} after initial write operation")
+        log.debug(obj_dump)
+        log.info(
+            "Verify the block size, total size and number of blocks after write operation"
+        )
+        try:
+            log.info(f"{obj_dump_stats['blksize']} == {bluestore_min_alloc_size}")
+            assert obj_dump_stats["blksize"] == bluestore_min_alloc_size
+            log.info(f"{obj_dump_stats['size']} == {iterations * 4 * 1048576}")
+            assert obj_dump_stats["size"] == (iterations * 4 * 1048576)
+            log.info(
+                f"{obj_dump_stats['blocks']} == {(iterations * 4 * 1048576) / bluestore_min_alloc_size}"
+            )
+            assert obj_dump_stats["blocks"] == int(
+                (iterations * 4 * 1048576) / bluestore_min_alloc_size
+            )
+            l_offset = obj_dump_extents[-1]["logical_offset"]
+            length = obj_dump_extents[-1]["length"]
+            log.info(f"{l_offset} + {length} == {(iterations * 4 * 1048576)}")
+            assert l_offset + length == (iterations * 4 * 1048576)
+            blocks_l_extent = int(
+                length / obj_dump_stats["blksize"]
+            )  # number of blocks in a logical extent
+            log.info(
+                f"{len(obj_dump_extents)} == {int(obj_dump_stats['blocks']/blocks_l_extent)}"
+            )
+            assert len(obj_dump_extents) == int(
+                obj_dump_stats["blocks"] / blocks_l_extent
+            )
+            log.info("PASS")
+        except (AssertionError, KeyError) as e:
+            log.info("^FAIL")
+            log.error(f"Failed with exception: {e.__doc__}")
+            log.exception(e)
+            raise
+
+        assert rados_obj.change_osd_state(action="start", target=acting_osd_id)
+        assert rados_obj.change_osd_state(action="enable", target=acting_osd_id)
+
+        for i in range(2):
+            time.sleep(3)
+            new_acting_pg_set = rados_obj.get_osd_map(pool=pool_name, obj=object_name)[
+                "acting"
+            ]
+            log.info(new_acting_pg_set)
+            if new_acting_pg_set == acting_pg_set:
+                break
+            elif i == 1:
+                log.error(
+                    f"{acting_osd_id} could not return to PG {new_acting_pg_set} within timeout. "
+                    f"New acting pg set({new_acting_pg_set} and old acting pg set({acting_pg_set}) are not same"
+                )
+                raise Exception("acting pg set has changed")
```

Comment:
Please set `nobackfill`, `norebalance`, `norecover`, `noout`, `nodown` . This should prevent the OSDs from being marked out, and the PG sets changing.

Reply:
Setting these flags did not help with RHCS 6.0 execution, and for RHCS 5.3 there flags are not required.